### DISCUSSION TITLE
SGD gradient calculation in Chapter 1,2,3

### DISCUSSION THREAD

**Discussion by surajbhv7l** [mrdbourke/pytorch-deep-learning#1034] [Q&A]
here is the code Snippet:

#setting the optimiser SGD
loss_fn = nn.CrossEntropyLoss()
optimizer = torch.optim.SGD(model_4.parameters(),
                            lr=0.1) # exercise: try changing the learning rate here and seeing what happens to the model's performance

#using SGD on full training data:
   loss = loss_fn(y_logits, y_blob_train)
    acc = accuracy_fn(y_true=y_blob_train,
                      y_pred=y_pred)

    # 3. Optimizer zero grad
    optimizer.zero_grad()

    # 4. Loss backwards
    loss.backward()

    # 5. Optimizer step
    optimizer.step()

My question is: Isn't the SGD here is acting as batch Gradient Descent as we are calculating gradient on full training data?

**Comment by LuluW8071** [comment]
If u don't set a batch size, it acts as Batch Gradient Descent (uses the entire dataset).
If u do set a batch size, it acts as Mini-Batch Gradient Descent (divides the dataset into batches and trains batch-wise)

Since u havent set batch_size in the code snippet, yes SGD acts as Batch Gradient Descent
